### PR TITLE
chore: Making automated version updates in readme.md

### PR DESCRIPTION
**Pull request by ddixit14**
This PR does 2 things:
1. Adding `readme_update.sh` script which is responsible for adding annotations in all the valid readme.md files.
2. After running the script, all the readme.md's have annotations in them, except a few modules.

Later on I will introduce how to use libraries-bom in all the readme's, along with automated version updates for it.

### CODE REVIEW

File: generation/readme_update.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```
Comment (suztomo) [inline generation/readme_update.sh:1]: When should this run? How about adding a check to ensure the README files have the annotations applied?

File: generation/readme_update.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+config_line=""
+
+for module in $(find . -mindepth 2 -maxdepth 2 -name pom.xml | sort | xargs dirname); do
+
+  if [[ "${module}" = *java-core ]] ||  [[ "${module}" = *java-shared-dependencies ]] ||  [[ "${module}" = *java-iam ]]; then
+      continue
```
Comment (suztomo) [inline generation/readme_update.sh:8]: Indent file.

File: generation/readme_update.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  fi
+
+  readme_file="${module}/README.md"
+
+  if [ -e ${module}/README.md ]; then
```
Comment (suztomo) [inline generation/readme_update.sh:13]: Indent file.

File: generation/readme_update.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+config_line=""
```
Comment (suztomo) [inline generation/readme_update.sh:3]: Typo?

File: generation/readme_update.sh (hunk 1 above)
Comment (ddixit14) [reply generation/readme_update.sh:1]: Explained in comments.

File: generation/readme_update.sh (hunk 2 above)
Comment (ddixit14) [reply generation/readme_update.sh]: done

File: generation/readme_update.sh (hunk 4 above)
Comment (ddixit14) [reply generation/readme_update.sh]: yes, removed now.

File: generation/readme_update.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+# This script should run as a part of new-library generation process.
```
Comment (suztomo) [inline generation/readme_update.sh:2]: Can you add corresponding change in generation/new_client in this pull request?

File: generation/readme_update.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+# This script introduces release-please annotations if they don't exist in the readme file
+
+for module in $(find . -mindepth 2 -maxdepth 2 -name pom.xml | sort | xargs dirname); do
+
+  if [[ "${module}" = *java-core ]] || [[ "${module}" = *java-shared-dependencies ]] || [[ "${module}" = *java-iam ]]; then
```
Comment (suztomo) [inline generation/readme_update.sh:7]: Should java-iam be excluded? If so, can you add source code comment why?

Comment (ddixit14) [reply generation/readme_update.sh]: I don't think it should have been excluded. I've added it back again, along with necessary changes in its readme.

File: generation/readme_update.sh (hunk 5 above)
Comment (ddixit14) [reply generation/readme_update.sh:2]: Done.

File: generation/readme_update.sh [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    artifactId=${new_string%"$suffix"}
+
+    echo "Handling ${artifactId}"
+
+    line_number=$(grep -n -m 1 "<dependency>" ${readme_file} | sed 's/\([0-9]*\).*/\1/')
+    start_line=${line_number}-2
+    line_number_end=$(grep -n -m 1 "Scala" ${readme_file} | sed 's/\([0-9]*\).*/\1/')
+    end_line=${line_number_end}+4
+
+    start_line_append="<!--- {x-version-update-start:${artifactId}:released} -->"
```
[the remark this answers is not in the document]
Comment (ddixit14) [reply generation/readme_update.sh:30]: yeah, it works either way.

[the remark this answers is not in the document]
Comment (suztomo) [reply generation/readme_update.sh:30]: Yes, it works but unnecessary. Let's not to have unnecessary characters.

**Review by suztomo — APPROVED** [review]

**Pull request merged**